crates/eips/src/eip4844/builder.rs

```diff
@@ -244,8 +249,21 @@ impl SidecarCoder for SimpleCoder {
     fn finish(self, _builder: &mut PartialSidecar) {}
 
     fn decode_all(&mut self, blobs: &[Blob]) -> Option<Vec<Vec<u8>>> {
-        let mut fes =
-            blobs.iter().flat_map(|blob| blob.chunks(32).map(WholeFe::new)).map(Option::unwrap);
+        if !matches!(blobs.len(), 1..=MAX_BLOBS_PER_BLOCK) {
+            return None;
+        }
```

Comment:
this restriction will be problematic soon, because this constant should no longer be used, so I'd rather remove this check entirely

Reply:
what's the problem? if the number of blobs is increased, we just update the constants and everything will work.

Reply:
the max number of blobs will be decoupled from this constant https://github.com/ethereum/EIPs/pull/9129

@klkvr should we use the `eip7840::BlobParams::cancun()` for now or remove this entirely, I guess the proper way of doing this would be to pass around the max_block setting but this seems a bit much...

Reply:
yeah don't think we need this check

custom implementations can store the constant in coder if needed

Reply:
Ok, I'll remove the use of `MAX_BLOBS_PER_BLOCK` in this place and here for now:
https://github.com/alloy-rs/alloy/blob/8ab278c57ed76a3fb37947bc1bbde62e5ee70d90/crates/eips/src/eip4844/builder.rs#L206-L209

Reply:
We could then do a check that `blobs: &[Blob]` is not empty and that `num_bytes` does not exceed 2 MiB.